src/docker.test.ts

```diff
@@ -47,32 +54,66 @@ describe("Docker images", (): void => {
   const assertLoadDockerImages = (cacheHit: boolean): void => {
     expect(core.getInput).lastCalledWith("key", { required: true });
     expect(cache.restoreCache).lastCalledWith([docker.DOCKER_IMAGES_PATH], KEY);
-    expect(core.saveState).lastCalledWith(docker.CACHE_HIT, cacheHit);
+    expect(core.saveState).nthCalledWith(1, docker.CACHE_HIT, cacheHit);
```

Comment:
Multiple instances: manually enforce type correctness on `nthCalledWith` calls within reason. You can see which cases we currently type and which we don't on main.